src/Http/Routing/ref/Microsoft.AspNetCore.Routing.netcoreapp.cs

```diff
@@ -90,6 +92,22 @@ public partial class EndpointNameMetadata : Microsoft.AspNetCore.Routing.IEndpoi
     }
     [System.AttributeUsageAttribute(System.AttributeTargets.Class | System.AttributeTargets.Method, AllowMultiple=false, Inherited=false)]
     [System.Diagnostics.DebuggerDisplayAttribute("{DebuggerToString(),nq}")]
+    public sealed partial class HeaderAttribute : System.Attribute, Microsoft.AspNetCore.Routing.IHeaderMetadata
+    {
+        public HeaderAttribute(string headerName, params string[] headerValues) { }
+        public string HeaderName { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } }
+        public System.Collections.Generic.IReadOnlyList<string> HeaderValues { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } }
+        public int MaximumValuesToInspect { get { throw null; } set { } }
```

Comment:
I don't think we need a special limit here, in most cases we are going to be matching a single header and if a header matches multiple values I think that's a semantic we should be happy with?

If you are concerned about the work the server has to do, the worst request that a client can send is only going to have a constant factor on the amount of work the server needs to do, so I think we are covered here based on the limits that the server already has in place for header size, so I think this optimizes for the uncommon case and that we shouldn't need it.

Does that make sense?